Address some more assorted smart pointer warnings in WebCore

annevk opened the pull request:
#### c243e2b5b3bfb8b1f788bfbf03ad09adaebb505a
<pre>
Address some more assorted smart pointer warnings in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=294900">https://bugs.webkit.org/show_bug.cgi?id=294900</a>

Reviewed by Chris Dumez.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a> to
some instances that were not flagged before in accessibility,
animation, pay, and some assorted code.

Canonical link: <a href="https://commits.webkit.org/296579@main">https://commits.webkit.org/296579@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7253a7a4696a8129bcd986f437574457c7e7dfc3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108956 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28616 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19041 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/114164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/59295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110919 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/29300 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/37184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/114164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/59295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111904 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/29300 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/98151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/114164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/29300 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/16286 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/58862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/29300 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/16334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117284 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36006 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/37184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/91805 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/36378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/94413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/91612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/14290 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31872 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17593 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/35904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/41429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/35605 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/38945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/37289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -553,7 +553,7 @@ FloatRect AccessibilityObject::convertFrameToSpace(const FloatRect& frameRect, A
 
     // Find the appropriate scroll view to use to convert the contents to the window.
-    const auto parentAccessibilityScrollView = ancestorAccessibilityScrollView(false /* includeSelf */);
+    const RefPtr parentAccessibilityScrollView = ancestorAccessibilityScrollView(false /* includeSelf */);
```
We don't need the `const`.

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3350,7 +3351,7 @@ bool AccessibilityObject::isExpanded() const
     // Summary element should use its details parent's expanded status.
     if (isSummary()) {
-        if (const AccessibilityObject* parent = Accessibility::findAncestor<AccessibilityObject>(*this, false, [] (const AccessibilityObject& object) {
+        if (const RefPtr parent = Accessibility::findAncestor<AccessibilityObject>(*this, false, [] (const AccessibilityObject& object) {
```
We don't need the `const`

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3550,8 +3551,8 @@ bool AccessibilityObject::isOnScreen() const
     for (size_t i = levels; i >= 1; i--) {
-        const AccessibilityObject* outer = objects[i];
-        const AccessibilityObject* inner = objects[i - 1];
+        const RefPtr outer = objects[i];
```
ditto

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3550,8 +3551,8 @@ bool AccessibilityObject::isOnScreen() const
-        const AccessibilityObject* outer = objects[i];
-        const AccessibilityObject* inner = objects[i - 1];
+        const RefPtr outer = objects[i];
+        const RefPtr inner = objects[i - 1];
```
ditto

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3651,8 +3652,8 @@ void AccessibilityObject::scrollToGlobalPoint(IntPoint&& point) const
     for (size_t i = 0; i < levels; i++) {
-        const AccessibilityObject* outer = objects[i];
-        const AccessibilityObject* inner = objects[i + 1];
+        const RefPtr outer = objects[i];
```
ditto

cdumez commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3651,8 +3652,8 @@ void AccessibilityObject::scrollToGlobalPoint(IntPoint&& point) const
-        const AccessibilityObject* outer = objects[i];
-        const AccessibilityObject* inner = objects[i + 1];
+        const RefPtr outer = objects[i];
+        const RefPtr inner = objects[i + 1];
```
ditto

cdumez commented on Source/WebCore/platform/animation/AcceleratedEffect.cpp:
```diff
@@ -172,7 +172,7 @@ static CSSPropertyID cssPropertyFromAcceleratedProperty(AcceleratedEffectPropert
 RefPtr<AcceleratedEffect> AcceleratedEffect::create(const KeyframeEffect& effect, const IntRect& borderBoxRect, const AcceleratedEffectValues& baseValues, OptionSet<AcceleratedEffectProperty>& disallowedProperties)
 {
-    auto* acceleratedEffect = new AcceleratedEffect(effect, borderBoxRect, disallowedProperties);
+    RefPtr acceleratedEffect = new AcceleratedEffect(effect, borderBoxRect, disallowedProperties);
```
Uh... no :) This should be:
```
RefPtr acceleratedEffect = adoptRef(new AcceleratedEffect(effect, borderBoxRect, disallowedProperties));
```

then you need to `return acceleratedEffect;` at the end of this function.

It is very unsafe to store a newly allocated object (with new) in a RefPtr without `adoptRef()` and would cause leaks in the early return case at line 178.

annevk replied:
Thanks for the review, should all be good now (modulo bots).

annevk committed:
Address some more assorted smart pointer warnings in WebCore
https://bugs.webkit.org/show_bug.cgi?id=294900

Reviewed by Chris Dumez.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines to
some instances that were not flagged before in accessibility,
animation, pay, and some assorted code.

Canonical link: https://commits.webkit.org/296579@main
